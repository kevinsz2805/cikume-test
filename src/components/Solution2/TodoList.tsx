import React from "react";
import { TodoProps, ItemsProps } from "../../interfaces/todo-props";

const TodoList = (props: TodoProps) => {
  return (
    <div className="page-content page-container" id="page-content">
      <div className="col-lg-12">
        <br />
      </div>
      <h2>TODO List</h2>

      <div className="form-check">
        {props.data.map((item: ItemsProps, index: number) => {
          return (
            <div key={item.id}>
              <input
                type="checkbox"
                className="form-check-input"
                value={index}
                checked={item.done}
                onChange={(e: any) => props.onChange(e, item)}
              />
              <label>{item.text}</label>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default TodoList;
