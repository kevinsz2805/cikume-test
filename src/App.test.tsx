import React from "react";
import TodoList from "./components/Solution2/TodoList";
import { render, fireEvent, waitForElement } from "@testing-library/react";

const mockData = [
  {
    id: 1,
    text: "Buy Grocery",
    done: false,
  },
  {
    id: 2,
    text: "Play Guitar",
    done: true,
  },
  {
    id: 3,
    text: "Romantic Dinner",
    done: false,
  },
];

test("rende <TodoList /> component", () => {
  const mockOnchange = jest.fn();
  const getWrapper = render(<TodoList data={[]} onChange={mockOnchange} />);
  expect(getWrapper).toMatchSnapshot();
});

test("render TODO List", () => {
  const mockOnchange = jest.fn();
  const getWrapper = render(<TodoList data={mockData} onChange={mockOnchange} />);
  expect(getWrapper).toMatchSnapshot();
});
