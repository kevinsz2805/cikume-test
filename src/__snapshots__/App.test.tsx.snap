// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rende <TodoList /> component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-content page-container"
        id="page-content"
      >
        <div
          class="col-lg-12"
        >
          <br />
        </div>
        <div
          class="col-lg-12"
        >
          <h2>
            TODO List
          </h2>
          <div
            class="form-check"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page-content page-container"
      id="page-content"
    >
      <div
        class="col-lg-12"
      >
        <br />
      </div>
      <div
        class="col-lg-12"
      >
        <h2>
          TODO List
        </h2>
        <div
          class="form-check"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render TODO List 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-content page-container"
        id="page-content"
      >
        <div
          class="col-lg-12"
        >
          <br />
        </div>
        <div
          class="col-lg-12"
        >
          <h2>
            TODO List
          </h2>
          <div
            class="form-check"
          >
            <div>
              <input
                class="form-check-input"
                type="checkbox"
                value="0"
              />
              <label>
                Buy Grocery
              </label>
            </div>
            <div>
              <input
                checked=""
                class="form-check-input"
                type="checkbox"
                value="1"
              />
              <label>
                Play Guitar
              </label>
            </div>
            <div>
              <input
                class="form-check-input"
                type="checkbox"
                value="2"
              />
              <label>
                Romantic Dinner
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page-content page-container"
      id="page-content"
    >
      <div
        class="col-lg-12"
      >
        <br />
      </div>
      <div
        class="col-lg-12"
      >
        <h2>
          TODO List
        </h2>
        <div
          class="form-check"
        >
          <div>
            <input
              class="form-check-input"
              type="checkbox"
              value="0"
            />
            <label>
              Buy Grocery
            </label>
          </div>
          <div>
            <input
              checked=""
              class="form-check-input"
              type="checkbox"
              value="1"
            />
            <label>
              Play Guitar
            </label>
          </div>
          <div>
            <input
              class="form-check-input"
              type="checkbox"
              value="2"
            />
            <label>
              Romantic Dinner
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
